### PR TITLE
Update android sdk 3.4.0

### PR DESCRIPTION
**Pull request by mickadoua**

### CODE REVIEW

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 1]
```diff
@@ -20,58 +23,88 @@ class RNPianoAnalyticsModule internal constructor(context: ReactApplicationConte
...
+    piano.init(context, config)
   }
 
   // EVENTS METHODS
   @ReactMethod
   override fun sendEvent(eventName: String, params: ReadableMap) {
-    piano.sendEvent(Event(eventName, params.toHashMap() as Map<String, String>))
+
+  // TODO : => il faut remplacer cette fonction par une boucle HashMap =>  old Method : piano.sendEvent(Event(eventName, params.toHashMap() as Map<String, String>))
+    PianoAnalytics.getInstance().sendEvents(
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: you can use **piano** instance you already defined line 17

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 2]
```diff
@@ -20,58 +23,88 @@ class RNPianoAnalyticsModule internal constructor(context: ReactApplicationConte
...
+        .build()
+    )
+
   }
 
   // USER METHODS
   @ReactMethod
   override fun setUser(id: String, category: String?, enableStorage: Boolean) {
-    piano.setUser(id, category, enableStorage)
+    val userStorage = PianoAnalytics.getInstance().userStorage
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt:60]: same comment for piano instance

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 3]
```diff
@@ -20,58 +23,88 @@ class RNPianoAnalyticsModule internal constructor(context: ReactApplicationConte
...
   override fun sendEvent(eventName: String, params: ReadableMap) {
-    piano.sendEvent(Event(eventName, params.toHashMap() as Map<String, String>))
+
+  // TODO : => il faut remplacer cette fonction par une boucle HashMap =>  old Method : piano.sendEvent(Event(eventName, params.toHashMap() as Map<String, String>))
+    PianoAnalytics.getInstance().sendEvents(
+      Event.Builder(eventName)
+        .properties(
+          Property(PropertyName("page"), "Page name"),
+          Property(PropertyName("page_premium"), false),
+          Property(PropertyName("page_authors"), arrayOf("Clark Kent","John Cena")),
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: maybe set pmutech for page authors

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 4]
```diff
@@ -20,58 +23,88 @@ class RNPianoAnalyticsModule internal constructor(context: ReactApplicationConte
...
+    val userStorage = PianoAnalytics.getInstance().userStorage
+    // set current user
+    userStorage.currentUser = User(id, category, enableStorage)
+  //  old Method : piano.setUser(id, category, enableStorage)
   }
 
   @ReactMethod
   override fun deleteUser() {
-    piano.deleteUser()
+    PianoAnalytics.getInstance()
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: you have already piano instance declared as **piano**

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 5]
```diff
@@ -20,58 +23,88 @@ class RNPianoAnalyticsModule internal constructor(context: ReactApplicationConte
...
+//    TODO : add function DELETE USER   sur la doc officiel il ny  pas de methode pour la version 3.3.0
+//  old Method :    piano.deleteUser()
   }
 
   // PRIVACY MODE
   @ReactMethod
   override fun privacySetMode(mode: String) {
-    piano.privacySetMode(mode)
+    PianoAnalytics.getInstance().privacyModesStorage
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: same here for piano instance

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 6]
```diff
@@ -20,58 +23,88 @@ class RNPianoAnalyticsModule internal constructor(context: ReactApplicationConte
...
+// DEPRECATED ?  utiliser consent a la place ?
+//    old Method : piano.privacySetMode(mode)
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: check this for privacyMode :

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 7]
```diff
@@ -10,12 +10,11 @@ import io.piano.android.analytics.model.Property
 import io.piano.android.analytics.model.PropertyName
 import io.piano.android.analytics.model.User
 
+
 @ReactModule(name = RNPianoAnalyticsModule.NAME)
-class RNPianoAnalyticsModule internal constructor(context: ReactApplicationContext) :
+class RNPianoAnalyticsModule internal constructor(var context: ReactApplicationContext) :
   RNPianoAnalyticsSpec(context) {
 
-  var piano = PianoAnalytics.getInstance();
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: why did you remove piano instance ?

File: android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt [hunk 8]
```diff
@@ -22,133 +23,111 @@ class RNPianoAnalyticsModule internal constructor(var context: ReactApplicationC
...
-    var eventNamesList: List<String>? = Arguments.toList(eventNames)?.map { it.toString() }
-    var eventNamesArray: Array<String> = eventNamesList?.toTypedArray() ?: emptyArray()
+  // PRIVACY INCLUDE PROPERTY
+  private fun includePropertiesInPrivacyModes(propertiesList: List<String>, privacyModesList: List<String>, eventNamesList: List<String>) {
+    for (privacyMode in privacyModesList) {
+      val mode = when (privacyMode) {
+        "exempt" -> PrivacyMode.EXEMPT
+        "optin" -> PrivacyMode.OPTIN
+        "optout" -> PrivacyMode.OPTOUT
+        else -> continue
```
Comment (ZHoussemEddine) [inline android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: you can refactor this code in a function duplicated on privacySetMode methode

Comment (mickadoua) [reply android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt]: cest fait
